pkg/api/datasources.go

```diff
@@ -54,7 +54,7 @@ func (hs *HTTPServer) GetDataSources(c *contextmodel.ReqContext) response.Respon
 		return response.Error(http.StatusInternalServerError, "Failed to query datasources", err)
 	}
 
-	filtered, err := hs.dsGuardian.New(c.SignedInUser.OrgID, c.SignedInUser).FilterDatasourcesByQueryPermissions(dataSources)
+	filtered, err := hs.dsGuardian.New(c.SignedInUser.OrgID, c.SignedInUser).FilterDatasourcesByReadPermissions(dataSources)
```

Comment:
I wonder if we need to make the same change [here](https://github.com/grafana/grafana/blob/main/pkg/api/frontendsettings.go#L402)?

Reply:
thanks, I remember seeing that but forgot to update it. https://github.com/grafana/grafana/pull/91345/commits/09999744333495bd19dbc82600300e026ec7db27